index.js

```diff
@@ -11,13 +13,134 @@ const FILE = 'pokedex.json';
 
 // Init express app
 const app = express();
+console.log("done creating app");
+
+// tell app to use the module
+app.use(express.json());
+app.use(express.urlencoded({
+    extended: true
+}));
+
+// method-override
+const methodOverride = require('method-override');
+app.use(methodOverride('_method'));
+
+// this line below, sets a layout look to your express project
+const reactEngine = require('express-react-views').createEngine();
+app.engine('jsx', reactEngine);
+
+// this tells express where to look for the view files
+app.set('views', __dirname + '/views');
+
+// this line sets react to be the default view engine
+app.set('view engine', 'jsx');
+
+// app.get('/', (req, res) => {
+//   // running this will let express to run home.handlebars file in your views folder
+//   res.render('home')
+// })
+
 
 /**
  * ===================================
  * Routes
  * ===================================
  */
 
+app.get('/', (request, response)=> {
+    const nameArr = [];
+    jsonfile.readFile(FILE, (err, obj) => {
+        for (i=0; obj['pokemon'].length; i++) {
```

Comment:
```suggestion
        const arrayLength = obj['pokemon'].length;
        for (i=0; i < arrayLength; i++) {
            const pokemon = obj['pokemon'][i];
            //add an if condition so push only works if the obj in the json file is NOT undefined or null
            if ( !(pokemon === null || pokemon === undefined) ) {
                nameArr.push(pokemon.name);
            }
        }
```
good practice to specify the array length outside the for-loop as a constant if you know are iterating over a fixed length
for-loop wasn't working properly due to missing "i <"
adding an if conditional to check for null/undefined entries in the object array helps prevent the code from breaking when bad entries in the json file are read
declaring the element being iterated at the beginning helps improve readability as we work with the element/object in the rest of the for-loop